salt/states/file.py

```diff
@@ -742,11 +742,17 @@ def _validate_str_list(arg):
     ensure ``arg`` is a list of strings
     '''
     if isinstance(arg, six.string_types):
-        return [arg]
+        ret = [arg]
     elif isinstance(arg, Iterable) and not isinstance(arg, Mapping):
-        return [str(item) for item in arg]
+        ret = []
+        for item in arg:
+            if isinstance(item, six.string_types):
+                ret.append(item)
+            else:
+                ret.append(str(item))
```

Comment:
why this change? is calling str() on every item too expensive, or does it have side effects we don't like? Mostly curious.